src/project.py

```diff
@@ -218,6 +233,52 @@ def get_tps_time(outfiles):
     return tps, f"{hh:02d}:{mm:02d}:{ss:02d}"
 
 
+@directives(ngpu=1)
+@on_pflow
+@MyProject.operation
+@MyProject.post(rdfed)
+@MyProject.pre(sampled)
+def post_proc(job):
+    import cmeutils
+    from cmeutils.structure import gsd_rdf
+    from cmeutils.structure import get_quaternions
+    import numpy as np
+    import os
+    import matplotlib
+    import matplotlib.pyplot as plt
+    import gsd
+    import gsd.hoomd
+    import freud
+
+    gsdfile= job.fn('trajectory.gsd')
+    rdf,norm = gsd_rdf(gsdfile,A_name='c', B_name='c', r_min=0.01, r_max=6)
+    x = rdf.bin_centers
+    y = rdf.rdf*norm
+    save_path= os.path.join(job.ws,"rdf.txt")
+    np.savetxt(save_path, np.transpose([x,y]), delimiter=',', header= "bin_centers, rdf")
+    plt.xlabel("r (A.U.)", fontsize=14)
+    plt.ylabel("g(r)", fontsize=14)
+    plt.plot(x, y)
+    save_plot= os.path.join(job.ws,"rdf.png")
+    plt.savefig(save_plot) 
+    with gsd.hoomd.open(gsdfile) as f:
+        snap = f[-1]
+        points = snap.particles.position
+        box = freud.Box.from_box(snap.configuration.box)
+        dp = freud.diffraction.DiffractionPattern(grid_size=1024, output_size=1024)
+        q_list= []
+        os.mkdir(os.path.join(job.ws,"diffraction_plots"))
```

Comment:
```suggestion
points = snap.particles.position
box = freud.Box.from_box(snap.configuration.box)
dp = freud.diffraction.DiffractionPattern(grid_size=1024, output_size=1024)
os.mkdir(os.path.join(job.ws,"diffraction_plots"))
```
we can move these outside of the `with` block